src/main/java/org/mskcc/oncokb/curation/service/CancerTypeService.java

```diff
@@ -128,6 +142,202 @@ public Optional<CancerType> findByMainTypeAndSubtypeIsNull(String mainType) {
         return cancerTypeRepository.findByMainTypeAndSubtypeIsNull(mainType);
     }
 
+    @Transactional(readOnly = true)
+    public List<CancerType> getRelevantCancerTypes(String levelOfEvidence, List<RelevantCancerTypeQuery> cancerTypeQueries) {
+        boolean isLevelBased = levelOfEvidence != null;
+        RelevantCancerTypeDirection direction = levelOfEvidence != null && levelOfEvidence.equals(LevelOfEvidenceEnum.LEVEL_DX1.getValue())
+            ? RelevantCancerTypeDirection.UPWARD
+            : RelevantCancerTypeDirection.DOWNWARD;
+        Set<CancerType> relevantCancerTypes = cancerTypeQueries
+            .stream()
+            .map(query -> {
+                if (StringUtils.isNullOrEmpty(query.getCode())) {
+                    if (isLevelBased) {
+                        List<CancerType> relevant = cancerTypeRepository.findAll();
+                        if (direction.equals(RelevantCancerTypeDirection.UPWARD)) {
```

Comment:
`direction` is guaranteed to be none null based on the context. But I typically use `RelevantCancerTypeDirection.UPWARD.equal(direction)`